src/_pytest/python.py

```diff
@@ -911,6 +911,124 @@ def hasnew(obj: object) -> bool:
     return False
 
 
+@final
+@attr.s(frozen=True, auto_attribs=True)
+class IdMaker:
+    argnames: Iterable[str] = attr.Factory(list)
+    parametersets: Iterable[ParameterSet] = attr.Factory(list)
+    idfn: Optional[Callable[[Any], Optional[object]]] = None
+    ids: Optional[List[Union[None, str]]] = None
+    config: Optional[Config] = None
+    nodeid: Optional[str] = None
+
+    def make_unique_test_ids(self) -> List[str]:
```

Comment:
Renamed to `make_unique_parameterset_ids`. But otoh, I also added a docstring that says `"Make a string for each ParameterSet to identify test invocations in pytest logs"`. In case the docstring is right, wouldn't it be more a "testinvocation_id" rather than a "parameterset_id"? Or in case the docstring is wrong, how is it wrong?

Reply:
`make_unique_parameterset_ids` makes perfect sense, sorry for the noise. I'll change the docstring.

It's `CallSpec2.id` that (again) joins `parameterset_ids` in `CallSpec2._idlist` (e.g. `['indirectparamval1', 'directparamval1-directparamval2']`) to the string that appears for a test invocation (like in `test_ids.py::test_foo[indirectparamval1-directparamval1-directparamval2] PASSED`). And even then, a "test invocation id" as suggested in my earlier commit would additionally have to include the test function name.